### DISCUSSION TITLE
YesSql DocumentId Generator Block Size Setting

### DISCUSSION THREAD

**Discussion by jeffolmstead** [OrchardCMS/OrchardCore#9866] [Q&A] [answered]
@jtkech seems like all the fun stuff I work on you had a hand it last implementing so I have copied you in as you might find this interesting. 

I am looking to increase the performance when I save a number of records at a time. I have gone through a SQL Profiler to see that a large part of the delay is that every 20 records it needs to retrieve the next Id. I narrowed this code down to here: https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs  

I am on SQL database and there is no block size passed in ([as seen on line 69](https://github.com/OrchardCMS/OrchardCore/blob/e2a1b00142fa7703de35ed57ab504cf0ff658952/src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs#L69)), it just uses the default (which I have checked over on YesSQL and the default is 20: https://github.com/sebastienros/yessql/blob/main/src/YesSql.Core/Services/DbBlockIdGenerator.cs)  In my perfect world there would be a tenant specific setting I can input to jump that up to 100 or more due to the nature of what I am building. Outside of that, I tried to modify it in a module via ISession like this:

```
_session.Store.Configuration.UseBlockIdGenerator(100);
```

But it throws an error that the "collection" was not initialized (which it isn't). I just don't know that rebuilding the store would even work as it is a singleton, nor where I would / could do it to make it tenant specific. Love to hear if you have any thoughts in this regard.

**Comment by jtkech** [comment, marked as answer]
@jeffolmstead Yes looks like there is no way to collaborate with the yesSql store `IConfiguration`.

Yes, here would be good to use `IShellConfiguration` to retrieve some config values, or resolve some `IOptions<StoreOptions>` so that we could set some values by code (e.g. `services.Configure<>()`), or both as we can also bind config values to options.

So for now i think you will have to override the `IStore` registration by copy pasting.

https://github.com/OrchardCMS/OrchardCore/blob/368b7825b75b3496980bf5d3d83515b3463df42e/src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs#L49

Your registration needs to be called after the one called by`.AddDataAccess()`, that's the case if done in a module startup. Then, you will just have to use the block size you want, or if you want retrieve it from some config / options as described above.
